ksql-engine/src/test/java/io/confluent/ksql/structured/SchemaKStreamTest.java

```diff
@@ -160,4 +167,45 @@ public void testSelectKey() throws Exception {
 
   }
 
+  @Test
+  public void testGroupByKey() {
+    String selectQuery = "SELECT col0, col1 FROM test1 WHERE col0 > 100;";
+    PlanNode logicalPlan = planBuilder.buildLogicalPlan(selectQuery);
+    initialSchemaKStream = new SchemaKStream(logicalPlan.getTheSourceNode().getSchema(), kStream,
+        ksqlStream.getKeyField(), new ArrayList<>(),
+        SchemaKStream.Type.SOURCE, functionRegistry, new MockSchemaRegistryClient());
+
+    Expression keyExpression = new DereferenceExpression(
+        new QualifiedNameReference(QualifiedName.of("TEST1")), "COL0");
+    KsqlTopicSerDe ksqlTopicSerDe = new KsqlJsonTopicSerDe();
+    Serde<GenericRow> rowSerde = ksqlTopicSerDe.getGenericRowSerde(
+        initialSchemaKStream.getSchema(), null, false, null);
+    List<Expression> groupByExpressions = Arrays.asList(keyExpression);
+    SchemaKGroupedStream groupedSchemaKStream = initialSchemaKStream.groupBy(
+        Serdes.String(), rowSerde, groupByExpressions);
+
+    Assert.assertEquals(groupedSchemaKStream.getKeyField().name(), "COL0");
```

Comment:
Out of curiosity: what would this assertion have been before the patch?

Reply:
This assertion wouldn't really make sense before this patch, since SchemaKStream::groupByKey assumed we were grouping by the key of the underlying stream. All these tests are doing right now is (indirectly) verifying that the resulting SchemaKStream is keyed by the right columns. Now that I think about it more, we should really be using a mock kstream to verify that the right groupBy... function is called. I'll update the test in another PR